core/src/main/java/org/apache/carbondata/core/datastore/chunk/store/impl/LocalDictDimensionDataChunkStore.java

```diff
@@ -94,10 +93,9 @@ public void fillVector(int[] invertedIndex, int[] invertedIndexReverse, byte[] d
   }
 
   @Override public void fillRow(int rowId, CarbonColumnVector vector, int vectorRow) {
-    if (!dictionary.isDictionaryUsed()) {
-      vector.setDictionary(dictionary);
-      dictionary.setDictionaryUsed();
-    }
+    // always set dictionary otherwise
+    // empty dictionary will get set if same col is called again in projection.
+    vector.setDictionary(dictionary);
```

Comment:
@BJangir 
1. Please check and confirm if the same problem occurs with CarbonSession also
2. Modify the PR description and specify the details for bug fixed in this PR after completion of point 1

Reply:
1. For Session it is ok, issue happens only in SDK reader , Now hanlded in org.apache.carbondata.hadoop.util.CarbonVectorizedRecordReader.  LocalDictDimensionDataChunkStore.java reverted back.
2. OK. Done